app/src/ui/changes/filter-changes-list.tsx

```diff
@@ -298,17 +270,66 @@ export class FilterChangesList extends React.Component<
     }
   )
 
+  /** Compute the 'Include All' checkbox value */
+  private getCheckAllValue = memoizeOne(
+    (
+      workingDirectory: WorkingDirectoryStatus,
+      rebaseConflictState: RebaseConflictState | null,
+      filteredItems: Map<string, IChangesListItem>
+    ) => {
+      if (
+        filteredItems.size === workingDirectory.files.length &&
+        rebaseConflictState === null
+      ) {
+        return getCheckBoxValueFromIncludeAll(workingDirectory.includeAll)
+      }
+
+      const files = workingDirectory.files.filter(f => filteredItems.get(f.id))
```

Comment:
Another minor nit but probably worth changing:
```suggestion
      const files = workingDirectory.files.filter(f => filteredItems.has(f.id))
```